lib/internal/modules/esm/translators.js

```diff
@@ -129,6 +131,11 @@ function createImportMetaResolve(defaultParentUrl) {
 
 function initializeImportMeta(meta, { url }) {
   // Alphabetical
+  if (StringPrototypeStartsWith(url, 'file:')) {
+    meta.__filename = internalURLModule.fileURLToPath(new URL(url));
+    meta.__dirname = internalPathModule.dirname(meta.__filename);
```

Comment:
can this be done with getters / lazily?

Reply:
Unfortunately not, there's a test that checks that `import.meta` properties are `writable`

Reply:
seems fine